src/native/libs/System.Security.Cryptography.Native/pal_evp_mac.c

```diff
@@ -0,0 +1,383 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "openssl.h"
+#include "pal_evp_mac.h"
+#include "pal_utilities.h"
+
+#include <assert.h>
+
+EVP_MAC* CryptoNative_EvpMacFetch(const char* algorithm, int32_t* haveFeature)
+{
+    assert(haveFeature);
+
+#ifdef NEED_OPENSSL_3_0
+    if (API_EXISTS(EVP_MAC_free))
+    {
+        *haveFeature = 1;
+        ERR_clear_error();
+        return EVP_MAC_fetch(NULL, algorithm, NULL);
+    }
+#else
+    (void)algorithm;
+    (void)haveFeature;
+#endif
+
+    *haveFeature = 0;
+    return NULL;
+}
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wmissing-noreturn"
```

Comment:
I would probably have just done assert+return since that's shorter and less thinky than the diagnostic disabling.  But this is fine.